src/server/game/Movement/MovementGenerator.h

```diff
@@ -28,22 +28,56 @@ class Unit;
 
 enum MovementGeneratorType : uint8;
 
+enum MovementGeneratorFlags : uint16
+{
+    MOVEMENTGENERATOR_FLAG_NONE                   = 0x000,
+    MOVEMENTGENERATOR_FLAG_INITIALIZATION_PENDING = 0x001,
+    MOVEMENTGENERATOR_FLAG_INITIALIZED            = 0x002,
+    MOVEMENTGENERATOR_FLAG_SPEED_UPDATE_PENDING   = 0x004,
+    MOVEMENTGENERATOR_FLAG_INTERRUPTED            = 0x008,
+    MOVEMENTGENERATOR_FLAG_PAUSED                 = 0x010,
+    MOVEMENTGENERATOR_FLAG_TIMED_PAUSED           = 0x020,
+    MOVEMENTGENERATOR_FLAG_DEACTIVATED            = 0x040,
+    MOVEMENTGENERATOR_FLAG_INFORM_ENABLED         = 0x080,
+    MOVEMENTGENERATOR_FLAG_FINALIZED              = 0x100,
+
+    MOVEMENTGENERATOR_FLAG_TRANSITORY = MOVEMENTGENERATOR_FLAG_SPEED_UPDATE_PENDING | MOVEMENTGENERATOR_FLAG_INTERRUPTED
+};
+
 class TC_GAME_API MovementGenerator
 {
     public:
+        MovementGenerator() : Mode(0), Priority(0), Flags(MOVEMENTGENERATOR_FLAG_NONE), BaseUnitState(0) { }
         virtual ~MovementGenerator();
 
+        // on top first update
         virtual void Initialize(Unit*) = 0;
-        virtual void Finalize(Unit*) = 0;
+        // on top reassign
         virtual void Reset(Unit*) = 0;
+        // on top on MotionMaster::Update
         virtual bool Update(Unit*, uint32 diff) = 0;
+        // on current top if another movement replaces
+        virtual void Deactivate(Unit*) = 0;
+        // on movement delete
+        virtual void Finalize(Unit*, bool, bool) = 0;
```

Comment:
Looks like travis-ci is unhappy with your use of `Finalize` here,
compared to the existing line `ChaseMovementGenerator.h:39`
https://github.com/TrinityCore/TrinityCore/blob/7e0c0e6b79f82e164b5270974418f2829fa8e982/src/server/game/Movement/MovementGenerators/ChaseMovementGenerator.h#L39
is this an include problem or does it need a different name?

Reply:
nah, I'm not done, still need to check chase generator